client/blocks/reader-post-card/featured-asset.jsx

```diff
@@ -2,37 +2,65 @@
  * External Dependencies
  */
 import React from 'react';
-import { head, filter, get } from 'lodash';
+import { find, endsWith, some, } from 'lodash';
+import url from 'url';
 
 /**
  * Internal Dependencies
  */
 import FeaturedVideo from './featured-video';
 import FeaturedImage from './featured-image';
 
+function isCandidateForFeature( media ) {
+	if ( ! media ) {
+		return false;
+	}
+
+	if ( media.mediaType === 'image' ) {
+		const image = media;
+
+		const imageIsTallEnough = 350 <= image.width || 350 <= image.naturalWidth;
+		const imageIsWideEnough = 85 <= image.height || 85 <= image.naturalHeight;
+
+		return imageIsTallEnough && imageIsWideEnough;
+	} else if ( media.mediaType === 'video' ) {
+		// we need to have a thumbnail and know how to autoplay it
+		return media.thumbnailUrl && media.autoplayIframe;
+	}
+
+	return true;
+}
+
+// jetpack lies about thumbnails so we need to make sure its actually an image. See: thumbIsLikelyImage in pick-canonical-image rule
+function isUrlLikelyAnImage( uri ) {
+	const withoutQuery = url.parse( uri ).pathname;
+	return some( [ '.jpg', '.jpeg', '.png', '.gif' ], ext => endsWith( withoutQuery, ext ) );
+}
+
+/**
+ * Given a post:
+ *  1. prefer to return the post's featured image ( post.post_thumbnail )
+ *  2. if there is no usable featured image, use the media that appears first in the content of the post
+ *  3. if there is no eligible asset, return null
+ */
 const FeaturedAsset = ( { post } ) => {
 	if ( ! post ) {
 		return null;
 	}
 
-	const featuredImage = post.canonical_image;
-	let featuredAsset;
-
-	// only feature an embed if we know how to thumbnail & autoplay it
-	const featuredEmbed = head( filter( post.content_embeds, ( embed ) => embed.thumbnailUrl && embed.autoplayIframe ) );
+	if ( post.featured_image && isUrlLikelyAnImage( post.featured_image ) ) {
```

Comment:
we need to check this for the min size too I think? Can use the dims on post_thumbnail.